Client/Frontend/Browser/GridTabViewController.swift

```diff
@@ -70,24 +73,88 @@ class GridTabViewController: UIViewController, TabTrayViewDelegate {
         return tabLayoutDelegate.numberOfColumns
     }
 
-    init(tabManager: TabManager, profile: Profile, tabTrayDelegate: TabTrayDelegate? = nil, tabToFocus: Tab? = nil) {
+    // MARK: - Inits
+    init(tabManager: TabManager,
+         profile: Profile,
+         tabTrayDelegate: TabTrayDelegate? = nil,
+         tabToFocus: Tab? = nil,
+         notificationCenter: NotificationCenter = NotificationCenter.default
+    ) {
         self.tabManager = tabManager
         self.profile = profile
         self.delegate = tabTrayDelegate
         self.tabToFocus = tabToFocus
+        self.notificationCenter = notificationCenter
+
+        let contextualViewModel = ContextualHintViewModel(forHintType: .inactiveTabs,
+                                                          with: profile)
+        self.contextualHintViewController = ContextualHintViewController(with: contextualViewModel)
+        
         super.init(nibName: nil, bundle: nil)
-        collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
-        collectionView.register(TabCell.self, forCellWithReuseIdentifier: TabCell.cellIdentifier)
-        collectionView.register(GroupedTabCell.self, forCellWithReuseIdentifier: GroupedTabCell.Identifier)
-        collectionView.register(InactiveTabCell.self, forCellWithReuseIdentifier: InactiveTabCell.Identifier)
-        collectionView.register(ASHeaderView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: GridTabViewController.independentTabsHeaderIdentifier)
-        tabDisplayManager = TabDisplayManager(collectionView: self.collectionView, tabManager: self.tabManager, tabDisplayer: self, reuseID: TabCell.cellIdentifier, tabDisplayType: .TabGrid, profile: profile)
+        collectionViewSetup()
+    }
+
+    private func collectionViewSetup() {
+        collectionView = UICollectionView(frame: .zero,
+                                          collectionViewLayout: UICollectionViewFlowLayout())
+        collectionView.register(cellType: TabCell.self)
+        collectionView.register(cellType: GroupedTabCell.self)
+        collectionView.register(cellType: InactiveTabCell.self)
+        collectionView.register(
+            ASHeaderView.self,
+            forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader,
+            withReuseIdentifier: GridTabViewController.independentTabsHeaderIdentifier)
```

Comment:
We could have an extension helper for this .register as well, in ReusableCell ?

Reply:
`ASHeaderView` does not conform to `ReusableCell` :)

Reply:
Ah, then when it does let's have a look at that. Future task then